lib/common/actions/index.js

```diff
@@ -42,43 +49,61 @@ export function secureFetch (url: string, method: string = 'get', payload?: any,
       .catch(err => {
-        const message = `Error making request: (${err})`
-        consoleError(message)
+        const message = `Error making ${method} request to ${url}`
+        console.error(message, err)
         return dispatch(setErrorMessage({message}))
       })
-      .then(res => {
-        // if raw response is requested
-        if (raw) return res
-        let action, message
-        // check for errors
-        const {status} = res
-        if (status >= 500) {
-          action = 'RELOAD'
-          message = `Network error!\n\n(${method} request on ${url})`
-          consoleError(message)
-          dispatch(setErrorMessage({message, action}))
-          return null
-        } else if (status >= 400) {
-          action = status === 401
-            ? 'LOG_IN'
-            : actionOnFail
-          res.json()
-            .then(json => {
-              const {detail, message} = json
-              const unknown = `Unknown (${status}) error occurred while making request`
-              consoleError(message || JSON.stringify(json) || unknown)
-              dispatch(setErrorMessage({
-                message: message || JSON.stringify(json) || unknown,
-                action,
-                detail
-              }))
-            })
-          return null
-        } else {
-          return res
-        }
-      })
+      .then(res => dispatch(handleResponse(method, url, res, raw)))
+  }
+}
+
+function handleResponse (method, url, res, raw, actionOnFail) {
+  return async function (dispatch: dispatchFn, getState: getStateFn) {
+    // if raw response is requested
+    if (raw) return res
+    const {status} = res
+    // Return response with no further action if there are no errors.
+    if (status < 400) return res
+    // check for errors
+    let json
+    try {
+      json = await res.json()
+    } catch (e) {
+      console.warn('Could not parse JSON from error response')
+    }
+    const errorMessage = getErrorMessageFromJson(json, status, method, url, actionOnFail)
+    dispatch(setErrorMessage(errorMessage))
+    return null
+  }
+}
+
+function getErrorMessageFromJson (
+  json: ?ErrorResponse,
+  status,
+  method,
+  url,
+  actionOnFail
+) {
+  let action = 'RELOAD'
+  let detail
+  let errorMessage = `Network error!\n\n(${method} request on ${url})`
+  if (status < 500) {
+    action = status === 401
+      ? 'LOG_IN'
+      : actionOnFail
+    errorMessage = `Unknown (${status}) error occurred while making request`
+  }
+  if (json) {
+    detail = json.detail
+    errorMessage = json.message || JSON.stringify(json)
+    // if status >= 500 and detail is being overrode, add original network error in small text
+    if (status >= 500) {
+      const networkMessage = `\nNetwork error!\n\n(${method} request on ${url})`
+      detail = detail ? detail + networkMessage : networkMessage
+    }
   }
+  console.error(`Error making ${method} request to ${url}: `, errorMessage)
```

Comment:
Use variable extracted from L52.

Reply:
Added variable in https://github.com/ibi-group/datatools-ui/pull/694/commits/7242f13da7192bc93fb1d537532a92cb68869981